docs/features/custom-license-and-keyword-searches.md

```diff
@@ -0,0 +1,211 @@
+
+```yaml
+version: 3
+
+customLicenseSearch:
+  - matchCriteria: (?i)this project is provided under a proprietary license
+    name: Proprietary License
+
+experimentalKeywordSearch:
+  - matchCriteria: abc123
+    name: Password
+```
+
+Both of these searches will run the regular expression provided in the `matchCriteria` field on every non-binary file in the directory that you are searching. The difference is in how the results are used.
+
+## Keyword Searches
+
+If a match to a keyword search is found, then the results of that search are output in the scan summary that `fossa analyze` outputs. For example, if you have just the `experimentalKeywordSearch` entry in the above `.fossa.yml` file and search a project that contains the string "abc123" in two files, then you will see something like this in the scan summary:
+
+```
+* Keyword Search: succeeded
+  ** Password - /Users/me/myproject/something.txt (lines 3-3)
+  ** Password - /Users/me/myproject/some/subdirectory/anotherfile.txt (lines 31-31)
+```
+
+## Custom License Searches
+
+If a match to a custom-license search is found, then the CLI will add the match to the licenses reported for the project being searched. The license will be identified as a "Custom License", and the name that you provided will be used in the FOSSA UI and in reports when displaying that license.
+
+The result will also be output in the scan summary, just as it is for [Keyword Searches](#keyword-searches).
+
+For example, if you have just the `customLicenseSearch` entry in the example `.fossa.yml` above and you find a match to the regular expression in two files, then the scan summary will look something like this:
+
+```
+* Custom-License Search: succeeded
+  ** Proprietary License - /Users/me/myproject/LICENSE (lines 4-4)
+  ** Proprietary License - /Users/me/myproject/src/main.rs (lines 2-3)
+```
+
+This will create a custom license match in your project in FOSSA. The results of that in the UI will look something like this:
+
+![A Custom License in the FOSSA UI](../assets/custom-license-scan-licenses-in-ui.png)
+
+The custom license will also be included in reports, and will look something like this:
+
+![A Custom License in a FOSSA Licensing report](../assets/custom-license-scan-report.png)
+
+Note that the name will be used as the license's name in the UI, so it is important to use names that are understandable to someone looking at license issues and viewing a report.
+
+If your project is set to raise issues for a license of type "Custom License", then an issue will be raised for any custom licenses found.
+
+## Regular expression format
+
+The regular expressions in Custom License and Keyword searches use Rust's regular expression syntax. Here are a few examples. You can also view the [full regular expression syntax documentation](./custom-license-and-keyword-search-regular-expression-syntax.md).
+
```

Comment:
This may be really obvious to folks used to worked with regexes, but I'd would add an explicit example calling out the fact that a simple string like "Match Me Exactly" is also a valid regular expression. Possibly repeat this where you first use the words "regular expression" and as a heading here.

You imply it in the first example, but I could see someone who hasn't worked with regexes before but wants to match something simple coming in and thinking they have to learn regex. Personally, I'd like customers to not use regex features as much as humanly possible in order to avoid support tickets that boil down to "Why doesn't my regex work?"

Reply:
Good idea! I added a section describing this, and also added some info on what characters you'll need to escape if you want to match them exactly